controllers/agentpool_controller.go

```diff
@@ -103,11 +102,6 @@ func (r *AgentPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	ap.log.Info("Agent Pool Controller", "msg", "successfully reconcilied agent pool")
 	r.Recorder.Eventf(&ap.instance, corev1.EventTypeNormal, "ReconcileAgentPool", "Successfully reconcilied agent pool ID %s", ap.instance.Status.AgentPoolID)
 
-	// Re-queue custom resource after the cool down period if applicable.
```

Comment:
I had to remove this as, we would have to compute the the current/desired replicas again here. I suspect a better approach here would be to refactor the functions that this `Reconcile()` is composed from can also return a `ctrl.Request`. Then we can return the computed requeue duration here: https://github.com/hashicorp/terraform-cloud-operator/blob/c3ff15e20aa1a345635e0a4df8c9ec1fc372afa6/controllers/agentpool_controller_autoscaling.go#L210-L213

A side effect of this change is that the scaledown period will have a lower bound of whatever the sync period has been configured to.